src/Emanote/Pandoc/Renderer/Embed.hs

```diff
@@ -43,14 +43,41 @@ embedWikiLinkResolvingSplice emaAction model nf (ctxSansCustomSplicing -> ctx) _
         B.Div (Url.brokenLinkAttr err) $
           one block
 
-embedSiteRoute :: Monad n => Ema.CLI.Action -> Model -> NoteRenderers n -> HP.RenderCtx n -> WL.WikiLink -> Either MN.Note SF.StaticFile -> Maybe (HI.Splice n)
-embedSiteRoute emaAction model nf RenderCtx {..} wl = \case
+embedInlineWikiLinkResolvingSplice ::
+  Monad n => PandocInlineRenderer n x
+embedInlineWikiLinkResolvingSplice emaAction model nf (ctxSansCustomSplicing -> ctx) _ inl =
+  case inl of
+    B.Link attr@(_id, _class, otherAttrs) is (url, tit) -> do
+      Rel.URTWikiLink (WL.WikiLinkEmbed, wl) <- Rel.parseUnresolvedRelTarget (otherAttrs <> one ("title", tit)) url
+      case Url.resolveWikiLinkMustExist model wl of
+        Left err ->
+          pure $ brokenLinkSpanWrapper err $ B.Link attr (Url.nonEmptyLinkInlines model url Nothing is) (url, tit)
+        Right res -> do
+          embedInlineSiteRoute emaAction model nf ctx wl res
+    _ ->
+      Nothing
+  where
+    brokenLinkSpanWrapper err inline =
+      HP.rpInline ctx $
+        B.Span (Url.brokenLinkAttr err) $ one inline
+
+embedBlockSiteRoute :: Monad n => Ema.CLI.Action -> Model -> NoteRenderers n -> HP.RenderCtx n -> WL.WikiLink -> Either MN.Note SF.StaticFile -> Maybe (HI.Splice n)
+embedBlockSiteRoute emaAction model nf RenderCtx {..} wl = \case
   Left note -> do
     pure . runEmbedTemplate "note" $ do
       "ema:note:title" ## Tit.titleSplice (preparePandoc model) (MN._noteTitle note)
       "ema:note:url" ## HI.textSplice (SR.siteRouteUrl model $ SR.lmlSiteRoute $ note ^. MN.noteRoute)
       "ema:note:pandoc"
         ## noteSpliceWith nf classMap emaAction model note
+  Right _ -> Nothing
+  where
+    runEmbedTemplate name splices = do
```

Comment:
It looks like `runEmbedTemplate ` is used in the other function as well? If so it should be made a top-level, an reused without duplication.